Backend/dev/migrations/dynamic-offer-driver-app/0157-refactor-fare-policy.sql

```diff
@@ -0,0 +1,166 @@
+
+WITH ProgressiveFarePolicy AS (
+  SELECT T1.id,
+    T1.base_distance_meters,
+    T1.base_distance_fare,
+    T1.per_extra_km_fare,
+    T1.dead_km_fare,
+    CASE
+        WHEN T1.waiting_charge_per_min IS NOT NULL
+        THEN REPLACE ('{"contents":???,"tag":"PerMinuteWaitingCharge"}' :: text, '???' :: text, T1.waiting_charge_per_min :: text) :: json
+	    ELSE NULL
+	END,
+    CASE
+        WHEN T1.night_shift_rate IS NOT NULL
+        THEN REPLACE ('{"contents":???,"tag":"ProgressiveNightShiftCharge"}' :: text, '???' :: text, T1.night_shift_rate :: text) :: json
+	    ELSE NULL
+	END
+  FROM atlas_driver_offer_bpp.fare_policy AS T1
+)
+INSERT INTO atlas_driver_offer_bpp.fare_policy_progressive_details (
+  fare_policy_id,
+  base_distance,
+  base_fare,
+  per_extra_km_fare,
+  dead_km_fare,
+  waiting_charge,
+  night_shift_charge)
+  (SELECT * FROM ProgressiveFarePolicy);
+
+ALTER TABLE atlas_driver_offer_bpp.fare_policy ALTER COLUMN driver_min_extra_fee DROP NOT NULL;
+ALTER TABLE atlas_driver_offer_bpp.fare_policy ALTER COLUMN driver_max_extra_fee DROP NOT NULL;
+
+ALTER TABLE atlas_driver_offer_bpp.fare_policy ADD COLUMN service_charge integer;
+ALTER TABLE atlas_driver_offer_bpp.fare_policy ADD COLUMN govt_charges float;
+ALTER TABLE atlas_driver_offer_bpp.fare_policy ADD COLUMN fare_policy_type character varying(50);
+UPDATE atlas_driver_offer_bpp.fare_policy AS T1 SET fare_policy_type = 'Progressive';
+ALTER TABLE atlas_driver_offer_bpp.fare_policy ALTER COLUMN fare_policy_type SET NOT NULL;
+
+ALTER TABLE atlas_driver_offer_bpp.fare_policy ALTER COLUMN base_distance_meters SET NOT NULL;
+ALTER TABLE atlas_driver_offer_bpp.fare_policy ALTER COLUMN base_distance_fare SET NOT NULL;
+ALTER TABLE atlas_driver_offer_bpp.fare_policy ALTER COLUMN per_extra_km_fare SET NOT NULL;
+ALTER TABLE atlas_driver_offer_bpp.fare_policy ALTER COLUMN dead_km_fare SET NOT NULL;
+ALTER TABLE atlas_driver_offer_bpp.fare_policy ALTER COLUMN waiting_charge_per_min SET NOT NULL;
+ALTER TABLE atlas_driver_offer_bpp.fare_policy ALTER COLUMN night_shift_rate SET NOT NULL;
+ALTER TABLE atlas_driver_offer_bpp.fare_policy ALTER COLUMN waiting_time_estimated_threshold SET NOT NULL;
+
+-- INSERT INTO atlas_driver_offer_bpp.fare_policy (
+--   T1.id,
+--   T1.merchant_id,
+--   T1.vehicle_variant,
+--   T1.fare_policy_type,
+--   T1.night_shift_start,
+--   T1.night_shift_end,
+--   T1.max_allowed_trip_distance,
+--   T1.min_allowed_trip_distance,
+--   T1.service_charge,
+--   T1.govt_charges)
+--   (SELECT
+--     (T1.id,
+--      T1.merchant_id,
+--      T1.vehicle_variant,
+--      'Slabs',
+--      T1.night_shift_start,
+--      T1.night_shift_end,
+--      T1.max_allowed_trip_distance,
+--      T1.min_allowed_trip_distance,
+--      T1.service_charge,
+--      T1.govt_charges)
+--     FROM atlas_driver_offer_bpp.slab_fare_policy AS T1 );
+-- Slabs are saved using show/read. It's rather difficult to migrate them
```

Comment:
Slab fare policy are not migrated. You have to migrate them manually or just create new fare policies for them